src/components/ToggleSwitch.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+
+const bgLight = `bg-[url('./assets/light.svg')]`
+const bgDark = `bg-[url('./assets/dark.svg')]`
+
+function ToggleSwitch({ isToggled, onChange }) {
+  const [isOn, setIsOn] = useState(isToggled)
+  const [bgPosition, setBgPosition] = useState('bg-left')
+  const [bgUrl, setBgUrl] = useState(bgLight)
+  const toggleHandler = () => {
+    setIsOn((oldState) => !oldState)
+  }
+
+  useEffect(() => {
+    onChange(isOn)
+    setBgPosition(isOn ? 'bg-left' : 'bg-right')
+    setBgUrl(isOn ? bgLight : bgDark)
```

Comment:
you can move this directly inside className, so that this would be changes along with state update, and there is no need to do this as side effect to state `isOn` change

Reply:
moved to className attribute